zo_compression_main.py

```diff
@@ -0,0 +1,217 @@
+from shared.language_utils import get_lm_loss, LM_TEMPLATE_MAP, SUPPORTED_LLM
+from shared.metrics import accuracy
+
+from tqdm import tqdm
+from gradient_estimators.random_gradient_estimator import RandomGradientEstimator as RGE
+from shared.quantized_layer import replace_layer
+
+
+def prepare_settings_underseed(args, device):
+    torch_dtype = {
+        "float32": torch.float32,
+        "float16": torch.float16,
+        "bfloat16": torch.bfloat16,
+    }[args.model_dtype]
+    torch.manual_seed(args.seed)
+    if args.dataset == "mnist":
+        model = CNN_MNIST().to(torch_dtype).to(device)
+        criterion = nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(
+            model.parameters(), lr=args.lr, weight_decay=1e-5, momentum=args.momentum
+        )
+        accuracy_func = accuracy
+        # scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.8)
+    elif args.dataset == "cifar10":
+        model = LeNet().to(torch_dtype).to(device)
+        criterion = nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(
+            model.parameters(), lr=args.lr, weight_decay=5e-4, momentum=args.momentum
+        )
+        accuracy_func = accuracy
+        # scheduler = torch.optim.lr_scheduler.MultiStepLR(
+        #     optimizer, milestones=[200], gamma=0.1
+        # )
+    elif args.dataset == "fashion":
+        model = CNN_FMNIST().to(torch_dtype).to(device)
+        criterion = nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(
+            model.parameters(), lr=args.lr, weight_decay=1e-5, momentum=args.momentum
+        )
+        accuracy_func = accuracy
+        # scheduler = torch.optim.lr_scheduler.MultiStepLR(
+        #     optimizer, milestones=[200], gamma=0.1
+        # )
+    elif args.dataset == "shakespeare":
+        model = CharLSTM().to(torch_dtype).to(device)
+        criterion = nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(model.parameters(), lr=args.lr, momentum=0.9, weight_decay=5e-4)
+        accuracy_func = accuracy
+        # scheduler = torch.optim.lr_scheduler.MultiStepLR(
+        #     optimizer, milestones=[200], gamma=0.1
+        # )
+    elif args.dataset in LM_TEMPLATE_MAP.keys():
+        large_model = args.large_model
+        model_name = SUPPORTED_LLM[large_model]
+        model = AutoModelForCausalLM.from_pretrained(model_name, torch_dtype=torch_dtype).to(device)
+        # Quantize the LLM
+        replace_layer(model)
+        model.model_name = large_model
+        tokenizer = AutoTokenizer.from_pretrained(
+            model_name, padding_side="left", truncate_side="left"
+        )
+        template = LM_TEMPLATE_MAP[args.dataset]()
+        verbalizer_id_map = template.get_verbalizer_id(tokenizer)
+        criterion = get_lm_loss("last_token", verbalizer_id_map)
+        optimizer = torch.optim.SGD(model.parameters(), lr=args.lr, momentum=0, weight_decay=5e-4)
+        accuracy_func = get_lm_loss("accuracy", verbalizer_id_map)
```

Comment:
just keep what is supported and delete the rest dataset.

Reply:
the original code is still in the `decom_fl_main.py`. It is always easy to copy-paste back when you are able to support compression on these models. It is always more time-consuming to debug unexpected behavior than copy-paste the code.